tests/test_core_network.py

```diff
@@ -1938,6 +1938,26 @@ def test_adding_multiple_links_with_id_and_multi_idx_clashes(assert_semantically
     )
 
 
+def test_adding_many_links_with_id_and_multi_idx_clashes(assert_semantically_equal, mocker):
```

Comment:
I know this isn't new in this PR, but using a PyTest fixture to pass in an assertion helper method is odd.

The PyTest fixture mechanism is for... well, test fixtures. A fixture is a concept in unit tests; it's the state or general environment needed to run the test. It's part of the setup. The helper method is not part of the test's fixture; it's part of the assertion or verification.

There are other, more obvious ways to make a helper assertion method available for the test. It feels like a misuse of the fixture mechanism to use it for things that are not fixtures.

Reply:
yeah that's really weird, I didn't even notice it (I accidentally copied it over from another test, I'm not using it here 😅 ) I think this change happened with the update to cookie cutter template etc (I remember things got moved to the conftest file around that time) but I don't remember why or how it ended up this way